AIDungeon_2.ipynb

```diff
@@ -36,49 +38,201 @@
         "print(\"Download Complete!\")"
-      ],
-      "execution_count": 0,
-      "outputs": []
+      ]
     },
     {
       "cell_type": "code",
+      "execution_count": 0,
       "metadata": {
-        "id": "YjArwbWh6XwN",
+        "colab": {},
         "colab_type": "code",
-        "colab": {}
+        "id": "YjArwbWh6XwN"
       },
+      "outputs": [],
       "source": [
+        "# Play\n",
         "from IPython.display import Javascript\n",
         "display(Javascript('''google.colab.output.setIframeHeight(0, true, {maxHeight: 5000})'''))\n",
         "!python play.py"
-      ],
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "kIldfwd8wjvT"
+      },
+      "source": [
+        "# Utilities (Persistent Save / Load, OOM Fix)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "zLx1yMu9wwBg"
+      },
+      "outputs": [],
+      "source": [
+        "# RUN THIS FIRST: Mount Google Drive\n",
+        "import pathlib\n",
+        "from distutils.dir_util import copy_tree\n",
+        "from google.colab import drive\n",
+        "\n",
+        "try:\n",
+        "  drive.mount('/content/drive')\n",
+        "except:\n",
+        "  pass\n",
+        "\n",
+        "drive_stories_directory=\"/content/drive/My Drive/AIDungeon/saved_stories\"\n",
+        "collab_stories_directory=\"/content/AIDungeon/saved_stories\"\n",
+        "\n",
+        "drive_model_directory=\"/content/drive/My Drive/Data/model_v5\"\n",
+        "collab_model_directory=\"/content/AIDungeon/generator/gpt2/models/model_v5\"\n",
+        "\n",
+        "pathlib.Path(drive_stories_directory).mkdir(parents=True, exist_ok=True) "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "LWfm6q8tAbDB"
+      },
+      "outputs": [],
+      "source": [
+        "# Save stories to Google Drive\n",
+        "copy_tree(\n",
+        "    collab_stories_directory, \n",
+        "    drive_stories_directory\n",
+        ")\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "HK2DO1jFxnv6"
+      },
+      "outputs": [],
+      "source": [
+        "# Load from Google Drive\n",
+        "copy_tree(\n",
+        "    drive_stories_directory, \n",
+        "    collab_stories_directory\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "XqK7MXhG40Oa"
+      },
+      "outputs": [],
+      "source": [
+        "# Copy model from Google Drive. Make sure the model is uploaded to your personal Drive. \n",
```

Comment:
Maybe this cell could go underneath the "backup model from collab" cell, so that newcomers put it into Google Drive before trying to load it from Google Drive?